core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -204,13 +204,21 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           mod1 >= mod2
       }
 
+      val tasks = mutable.ListBuffer.empty[Future[_]]
       logInfos.sliding(20, 20).foreach { batch =>
```

Comment:
you can avoid all mutability with

```scala
val tasks: Seq[Future[_]] = logInfos.grouped(20).map{ batch =>
  replayExecutor.submit(new Runnable {
    override def run(): Unit = mergeApplicationListing(batch)
  })
}
```

(I know the change to `grouped` is unrelated, but everytime I look at this code it confuses me for a second why we have overlapping sliding windows -- might as well as clean it up while you are messing around here)